third_party/nvfuser/csrc/expr_simplifier.h

```diff
@@ -564,8 +564,13 @@ RegisterType getRegisterType(Val* value);
 // scalars, regardless of if it is inside `variables` or not.
 // See note [Reordering associative and commutative operators] for detailed
 // information about this reordering.
+//
+// If `preserve_error` is false, we do simplification like a*b/b -> a unless we
```

Comment:
I wonder what implications  we would have with `preserve_error = false`. I don't have a good understanding of how critical preserving these errors would be. What do you think?

Maybe helpful to have some (optional) logging of potentially unsafe simplifications?

Reply:
I actually think `preserve_error = false` is a good thing because this might improve our zero tensor support. I didn't get deep but I do remember seeing index like `i / size * size + i % size`, for this, if we simplify it to `i`, then likely we will have zero tensor support automatically.

It is definitely helpful to add optional logging of this.

Reply:
debug dump added